fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(todos => this.setState({
+        todos
+      }));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(todo => {
+        this.state.todos.push(todo);
+        return this.setState({
+          todos: this.state.todos
+        });
+      });
+
+  }
+
+  toggleCompleted(todoId) {
+    const todoList = this.state.todos;
+    const result = todoList.map(todo => {
+      if (todo.todoId === todoId) {
+        return { isCompleted: todo.isCompleted };
+      }
+    });
+    fetch(`/api/todos/${todoId}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(result[todoId - 1])
```

Comment:
Ok, be suuuuper careful not to assumed that indexes have anything to do with ids. They are not related concepts so this subtraction doesn't make any sense.